addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command
+from odoo import models, _
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
+        self.ensure_one()
+        lines = super(SaleOrder, self)._cart_find_product_line(product_id, line_id)
+        if line_id:
+            return lines
+        event_booth_pending_ids = self.env.context.get('event_booth_pending_ids')
+        if event_booth_pending_ids:
+            lines = lines.filtered(
+                lambda line: any(booth.id in event_booth_pending_ids for booth in line.event_booth_pending_ids)
+            )
+        return lines
+
+    def _website_product_id_change(self, order_id, product_id, qty=0):
```

Comment:
That's ... well. I suppose you have no choice but that quite a ping-pong coding style. Anyway I don't understand this code. Will have a look again maybe tomorrow but... hey ...